commonmark_extensions/tables.py

```diff
@@ -25,7 +25,8 @@ def table(parser, container):
             parser.advance_offset(1, False)
             parser.close_unmatched_blocks()
             parser.add_child('table', parser.next_nonspace)
-            return 1
+            parser.tip.string_content = '|'
+            return 2
```

Comment:
Returning 2 makes it a leaf node, which in practice means that it doesn't slurp up whitespace after the `|`, meaning the pipe can be added back simply.